src/rt/sched/cown.h

```diff
@@ -52,18 +52,41 @@ namespace verona::rt
 
   /**
    * A cown, or concurrent owner, encapsulates a set of resources that may be
-   * accessed by a single (scheduler) thread at a time. A cown can only be in
-   * exactly one of the following states:
+   * accessed by a single (scheduler) thread at a time when writing, or
+   * accessed by multiple (scheduler) threads at a time when reading.
+   * A cown can only be in one of the following states:
    *   1. Unscheduled
    *   2. Scheduled, in the queue of a single scheduler thread
-   *   3. Running on a single scheduler thread
+   *   3. Running with read/write access on a single scheduler thread, and not
+   * in the queue of any scheduler thread
+   *   4. Running with read access on one or more scheduler threads, and may
+   * also be in the queue of one other scheduler thread
    *
    * Once a cown is running, it executes a batch of multi-message behaviours.
    * Each message may either acquire the running cown for participation in a
    * future behaviour, or execute the behaviour if it is the last cown to be
-   * acquired. If the running cown is acquired for a future behaviour, it will
-   * be descheduled until that behaviour has completed.
+   * acquired.
+   * If the running cown is acquired for writing for a future behaviour, it will
+   * be descheduled until that behaviour has completed. If the running cown is
+   * acquired for reading for a future behaviour, it will not be descheduled. If
+   * the running cown is acquired for reading _and_ executing on this thread,
+   * the cown will be rescheduled to be picked up by another thread. (it might
+   * later return to this thread if this is the last thread to use the cown in
+   * read more before a write).
    */
+
+  enum class AccessMode
+  {
+    WRITE,
+    READ
+  };
+
+  struct Request
+  {
```

Comment:
Could you make this more abstract with accessors, so that we can use the bottom bit of the pointer for the mode.  You don't need to do the bit packing, but make a class that could be changed without affecting anything else.